test/functional/rpc_fundrawtransaction.py

```diff
@@ -708,74 +707,108 @@ def test_all_watched_funds(self):
         for mode in ["foo", Decimal("3.141592")]:
             assert_raises_rpc_error(-8, "Invalid estimate_mode parameter",
-                lambda: self.nodes[1].fundrawtransaction(rawtx, {"estimate_mode": mode, "conf_target": 0.1}))
+                node.fundrawtransaction, rawtx, {"estimate_mode": mode, "conf_target": 0.1})
 
         self.log.info("Test fundrawtxn with invalid conf_target settings")
-        for mode in ["unset", "economical", "conservative", "btc/kb", "sat/b"]:
+        for mode in ["unset", "economical", "conservative"]:
             self.log.debug("{}".format(mode))
             for k, v in {"string": "", "object": {"foo": "bar"}}.items():
                 assert_raises_rpc_error(-3, "Expected type number for conf_target, got {}".format(k),
-                    lambda: self.nodes[1].fundrawtransaction(rawtx, {"estimate_mode": mode, "conf_target": v}))
-            if mode in ["btc/kb", "sat/b"]:
-                assert_raises_rpc_error(-3, "Amount out of range",
-                    lambda: self.nodes[1].fundrawtransaction(rawtx, {"estimate_mode": mode, "conf_target": -1}))
-                assert_raises_rpc_error(-4, "Fee rate (0.00000000 BTC/kB) is lower than the minimum fee rate setting (0.00001000 BTC/kB)",
-                    lambda: self.nodes[1].fundrawtransaction(rawtx, {"estimate_mode": mode, "conf_target": 0}))
-            else:
-                for n in [-1, 0, 1009]:
-                    assert_raises_rpc_error(-8, "Invalid conf_target, must be between 1 and 1008",
-                        lambda: self.nodes[1].fundrawtransaction(rawtx, {"estimate_mode": mode, "conf_target": n}))
-
-        for unit, fee_rate in {"sat/B": 0.99999999, "BTC/kB": 0.00000999}.items():
-            self.log.info("- raises RPC error 'fee rate too low' if conf_target {} and estimate_mode {} are passed".format(fee_rate, unit))
-            assert_raises_rpc_error(-4, "Fee rate (0.00000999 BTC/kB) is lower than the minimum fee rate setting (0.00001000 BTC/kB)",
-                lambda: self.nodes[1].fundrawtransaction(rawtx, {"estimate_mode": unit, "conf_target": fee_rate, "add_inputs": True}))
+                    node.fundrawtransaction, rawtx, {"estimate_mode": mode, "conf_target": v})
+            for n in [-1, 0, 1009]:
+                assert_raises_rpc_error(-8, "Invalid conf_target, must be between 1 and 1008",
+                    node.fundrawtransaction, rawtx, {"estimate_mode": mode, "conf_target": n})
+
+        self.log.info("Test invalid fee rate settings")
+        assert_raises_rpc_error(-4, "Fee rate (0.00000000 BTC/kB) is lower than the minimum fee rate setting (0.00001000 BTC/kB)",
+            node.fundrawtransaction, rawtx, {"fee_rate": 0, "add_inputs": True})
+        assert_raises_rpc_error(-8, "Invalid fee_rate 0.00000000 BTC/kB (must be greater than 0)",
+            node.fundrawtransaction, rawtx, {"feeRate": 0, "add_inputs": True})
+        for param, value in {("fee_rate", 100000), ("feeRate", 1.1)}:
```

Comment:
I'm not sure if we did that already, but it would be good to assure that the valid value range for `fee_rate` will cause an error for `feeRate` and vice versa. I would expect a lot of people to plug values for either into the other, and I'm not sure off the top of my head that the ranges have no overlap. (They should be shifted by a factor of 1e5 though, right?)

Reply:
Good point. The ranges don't overlap if I'm not confused, except in the minumum fee direction. We already check the min/max for fee_rate (0.1-100,000 sat/vB) and the max for feeRate (1 BTC/kvB) in the functional tests. At the moment, min fee checks with feeRate are disabled due to `coinControl.fOverrideFeeRate = true` in wallet/rpcwallet.cpp `FundTransaction()`.

To test the minimum for feeRate (0.00000999 BTC/kvB), we'd have to remove that `fOverrideFeeRate` setting. I've added this for now:

```diff
test/functional/rpc_fundrawtransaction.py
@@ 772 @@ class RawTransactionsTest(BitcoinTestFramework):
+        # This test will only pass if `coinControl.fOverrideFeeRate = true` in wallet/rpcwallet.cpp::FundTransaction is removed
+        assert_raises_rpc_error(-4, "Fee rate (0.999 sat/vB) is lower than the minimum fee rate setting (1.000 sat/vB)",
+            node.fundrawtransaction, rawtx, {"feeRate": 0.00000999, "add_inputs": True})
```

Reply:
It's not clear to me if `fOverrideFeeRate` should be added to `fee_rate` in fundrawtransaction as well, or just removed. IIUC it currently only serves to not check for a minimum `feeRate` in fundrawtransaction only.

Reply:
It's not clear to me if `fOverrideFeeRate` should be set to true if `fee_rate` is passed in fundrawtransaction or walletcreatefundedpsbt, as it is for `feeRate`, or if the `fOverrideFeeRate` flag could just be removed.

AFAICT `fOverrideFeeRate` set to true currently only serves to bypass minimum feerate checks for `feeRate` in `GetMinimumFeeRate()` for fundrawtransaction and walletcreatefundedpsbt.

Reply:
Found this [comment](https://github.com/bitcoin/bitcoin/pull/10706/#discussion_r126569510) in #10706 that describes the reason in mid-2017 for `fOverrideFeeRate` pertaining to `feeRate` in fundrawtxn (and now wallcreatefundedpsbt).

Reply:
Found this [mini-discussion](https://github.com/bitcoin/bitcoin/pull/10706/#discussion_r126560525) in #10706 in mid-2017 between @ryanofsky and @morcos about the use of `fOverrideFeeRate` for `feeRate` in fundrawtransaction (and now also walletcreatefundedpsbt). Thoughts WRT to using `fOverrideFeeRate` for the new `fee_rate` in these two RPCs -- yay or nay?

Reply:
Decided to maintain the `fOverrideFeeRate` behavior of disabling the min fee checks on `fee_rate` in RPCs fundrawtransaction and walletcreatefundedpsbt only, as has been the case for `feeRate` in these two RPCs since 2017, per the above comments. Done in afe94710ef8. Resolving.